File: frontend/src/components/AuthForm.js
// // import React, { useState } from 'react';
// // import axios from 'axios';

// // const AuthForm = () => {
// //   const [isLogin, setIsLogin] = useState(true);
// //   const [form, setForm] = useState({ name: '', email: '', phone: '' });

// //   const handleChange = (e) => {
// //     setForm({ ...form, [e.target.name]: e.target.value });
// //   };

// //  const handleSubmit = async (e) => {
// //   e.preventDefault();

// //   const baseUrl = process.env.REACT_APP_API_URL;
// //   const url = `${baseUrl}/api/auth/${isLogin ? 'login' : 'signup'}`;

// //   // ✅ Normalize the phone format before sending
// //   let phone = form.phone.trim();

// //   // Add country code if missing
// //   if (!phone.startsWith('+')) {
// //     phone = '+91' + phone;
// //   }

// //   const payload = isLogin
// //     ? { phone }
// //     : { name: form.name, email: form.email, phone };

// //   try {
// //     const res = await axios.post(url, payload);
// //     alert(res.data.msg);
// //   } catch (err) {
// //     alert(err.response?.data?.msg || 'Error');
// //   }
// // };


// //   return (
// //     <div style={styles.container}>
    
// //       <h2>{isLogin ? 'Login' : 'Signup'} - Surgy Innovations Lab</h2>
      
// //       <form onSubmit={handleSubmit} style={styles.form}>
// //         {!isLogin && (
// //           <>
// //             <input
// //               type="text"
// //               name="name"
// //               placeholder="Name"
// //               value={form.name}
// //               onChange={handleChange}
// //               required
// //               style={styles.input}
// //             />
// //             <input
// //               type="email"
// //               name="email"
// //               placeholder="Email"
// //               value={form.email}
// //               onChange={handleChange}
// //               required
// //               style={styles.input}
// //             />
// //           </>
// //         )}
// //         <input
// //           type="tel"
// //           name="phone"
// //           placeholder="Phone (e.g., +91XXXXXXXXXX)"
// //           value={form.phone}
// //           onChange={handleChange}
// //           required
// //           style={styles.input}
// //         />
// //         <button type="submit" style={styles.button}>
// //           {isLogin ? 'Login' : 'Signup'}
// //         </button>
// //       </form>
// //       <button onClick={() => setIsLogin(!isLogin)} style={styles.toggleButton}>
// //         Switch to {isLogin ? 'Signup' : 'Login'}
// //       </button>
// //     </div>
// //   );
// // };

// // const styles = {
// //   container: {
// //     width: '100%',
// //     maxWidth: '400px',
// //     margin: 'auto',
// //     padding: '20px',
// //     textAlign: 'center',
// //     backgroundColor: '#f0f4f8',
// //     borderRadius: '10px',
// //     boxShadow: '0 0 15px rgba(0,0,0,0.1)',
// //     fontFamily: 'Arial, sans-serif',
// //   },
// //   form: {
// //     display: 'flex',
// //     flexDirection: 'column',
// //     gap: '15px',
// //   },
// //   input: {
// //     padding: '10px',
// //     fontSize: '16px',
// //     borderRadius: '5px',
// //     border: '1px solid #ccc',
// //   },
// //   button: {
// //     backgroundColor: '#4CAF50',
// //     color: '#fff',
// //     padding: '10px',
// //     border: 'none',
// //     borderRadius: '5px',
// //     fontSize: '16px',
// //     cursor: 'pointer',
// //   },
// //   toggleButton: {
// //     marginTop: '10px',
// //     backgroundColor: 'transparent',
// //     color: '#007BFF',
// //     border: 'none',
// //     cursor: 'pointer',
// //     fontSize: '14px',
// //   },
// // };

// // export default AuthForm;
// import React, { useState } from 'react';
// import axios from 'axios';

// const AuthForm = () => {
//   const [isLogin, setIsLogin] = useState(true);
//   const [form, setForm] = useState({ name: '', email: '', phone: '' });

//   const handleChange = (e) => {
//     setForm({ ...form, [e.target.name]: e.target.value });
//   };

//   const handleSubmit = async (e) => {
//     e.preventDefault();

//     const baseUrl = process.env.REACT_APP_API_URL;
//     const url = `${baseUrl}/api/auth/${isLogin ? 'login' : 'signup'}`;

//     let phone = form.phone.trim();
//     if (!phone.startsWith('+')) {
//       phone = '+91' + phone;
//     }

//     const payload = isLogin
//       ? { phone }
//       : { name: form.name, email: form.email, phone };

//     try {
//       const res = await axios.post(url, payload);
//       alert(res.data.msg);
//     } catch (err) {
//       alert(err.response?.data?.msg || 'Error');
//     }
//   };
  

//   return (
//     <div style={styles.container}>
//       <h2>{isLogin ? 'Login' : 'Signup'} - Surgy Innovations Lab</h2>

      
//         <div style={styles.whatsappBox}>
//           <p style={styles.whatsappText}>
//             📢 <strong>Before signing up:</strong> To receive WhatsApp alerts from Surgy Innovations Labs, you must register your number.
//           </p>
//           <a
//             href="https://wa.me/14155238886?text=join%20eager-forest" // replace with your join code
//             target="_blank"
//             rel="noopener noreferrer"
//             style={styles.whatsappButton}
//           >
//             ✅ Click here to register on WhatsApp
//           </a>
//         </div>
      

//       <form onSubmit={handleSubmit} style={styles.form}>
//         {!isLogin && (
//           <>
//             <input
//               type="text"
//               name="name"
//               placeholder="Name"
//               value={form.name}
//               onChange={handleChange}
//               required
//               style={styles.input}
//             />
//             <input
//               type="email"
//               name="email"
//               placeholder="Email"
//               value={form.email}
//               onChange={handleChange}
//               required
//               style={styles.input}
//             />
//           </>
//         )}
//         <input
//           type="tel"
//           name="phone"
//           placeholder="Phone (e.g., +91XXXXXXXXXX)"
//           value={form.phone}
//           onChange={handleChange}
//           required
//           style={styles.input}
//         />
//         <button type="submit" style={styles.button}>
//           {isLogin ? 'Login' : 'Signup'}
//         </button>
//       </form>
//       <button onClick={() => setIsLogin(!isLogin)} style={styles.toggleButton}>
//         Switch to {isLogin ? 'Signup' : 'Login'}
//       </button>
//     </div>
//   );
// };

// const styles = {
//   container: {
//     width: '100%',
//     maxWidth: '400px',
//     margin: 'auto',
//     padding: '20px',
//     textAlign: 'center',
//     backgroundColor: '#f0f4f8',
//     borderRadius: '10px',
//     boxShadow: '0 0 15px rgba(0,0,0,0.1)',
//     fontFamily: 'Arial, sans-serif',
//   },
//   form: {
//     display: 'flex',
//     flexDirection: 'column',
//     gap: '15px',
//   },
//   input: {
//     padding: '10px',
//     fontSize: '16px',
//     borderRadius: '5px',
//     border: '1px solid #ccc',
//   },
//   button: {
//     backgroundColor: '#4CAF50',
//     color: '#fff',
//     padding: '10px',
//     border: 'none',
//     borderRadius: '5px',
//     fontSize: '16px',
//     cursor: 'pointer',
//   },
//   toggleButton: {
//     marginTop: '10px',
//     backgroundColor: 'transparent',
//     color: '#007BFF',
//     border: 'none',
//     cursor: 'pointer',
//     fontSize: '14px',
//   },
//   whatsappBox: {
//     marginBottom: '20px',
//     backgroundColor: '#e6f7f1',
//     border: '1px solid #25d366',
//     borderRadius: '8px',
//     padding: '12px',
//   },
//   whatsappText: {
//     marginBottom: '10px',
//     color: '#333',
//     fontSize: '14px',
//   },
//   whatsappButton: {
//     backgroundColor: '#25d366',
//     color: 'white',
//     padding: '10px 15px',
//     textDecoration: 'none',
//     borderRadius: '6px',
//     fontWeight: 'bold',
//     display: 'inline-block',
//   },
// };

// export default AuthForm;
import React, { useState } from 'react';
import axios from 'axios';

const AuthForm = () => {
  const [isLogin, setIsLogin] = useState(true);
  const [isLoggedIn, setIsLoggedIn] = useState(false); // ✅ new state
  const [form, setForm] = useState({ name: '', email: '', phone: '' });

  const handleChange = (e) => {
    setForm({ ...form, [e.target.name]: e.target.value });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();

    const baseUrl = process.env.REACT_APP_API_URL;
    const url = `${baseUrl}/api/auth/${isLogin ? 'login' : 'signup'}`;

    let phone = form.phone.trim();
    if (!phone.startsWith('+')) {
      phone = '+91' + phone;
    }

    const payload = isLogin
      ? { phone }
      : { name: form.name, email: form.email, phone };

    try {
      const res = await axios.post(url, payload);
      alert(res.data.msg);
      
      if (isLogin) {
        // ✅ switch screen after login success
        setIsLoggedIn(true);
      }
    } catch (err) {
      alert(err.response?.data?.msg || 'Error');
    }
  };

  if (isLoggedIn) {
    return (
      <div style={styles.container}>
        <h1>🎉 Welcome to Surgy Innovation Labs</h1>
        <p>We're excited to have you on board.</p>
      </div>
    );
  }

  return (
    <div style={styles.container}>
      <h2>{isLogin ? 'Login' : 'Signup'} - Surgy Innovations Lab</h2>

      {!isLogin && (
        <div style={styles.whatsappBox}>
          <p style={styles.whatsappText}>
            📢 <strong>Before signing up:</strong> To receive WhatsApp alerts from Surgy Innovations Labs, you must register your number.
          </p>
          <a
            href="https://wa.me/14155238886?text=join%20eager-forest"
            target="_blank"
            rel="noopener noreferrer"
            style={styles.whatsappButton}
          >
            ✅ Click here to register on WhatsApp
          </a>
        </div>
      )}

      <form onSubmit={handleSubmit} style={styles.form}>
        {!isLogin && (
          <>
            <input
              type="text"
              name="name"
              placeholder="Name"
              value={form.name}
              onChange={handleChange}
              required
              style={styles.input}
            />
            <input
              type="email"
              name="email"
              placeholder="Email"
              value={form.email}
              onChange={handleChange}
              required
              style={styles.input}
            />
          </>
        )}
        <input
          type="tel"
          name="phone"
          placeholder="Phone (e.g., +91XXXXXXXXXX)"
          value={form.phone}
          onChange={handleChange}
          required
          style={styles.input}
        />
        <button type="submit" style={styles.button}>
          {isLogin ? 'Login' : 'Signup'}
        </button>
      </form>
      <button onClick={() => setIsLogin(!isLogin)} style={styles.toggleButton}>
        Switch to {isLogin ? 'Signup' : 'Login'}
      </button>
    </div>
  );
};

const styles = {
  container: {
    width: '100%',
    maxWidth: '400px',
    margin: 'auto',
    padding: '20px',
    textAlign: 'center',
    backgroundColor: '#f0f4f8',
    borderRadius: '10px',
    boxShadow: '0 0 15px rgba(0,0,0,0.1)',
    fontFamily: 'Arial, sans-serif',
  },
  form: {
    display: 'flex',
    flexDirection: 'column',
    gap: '15px',
  },
  input: {
    padding: '10px',
    fontSize: '16px',
    borderRadius: '5px',
    border: '1px solid #ccc',
  },
  button: {
    backgroundColor: '#4CAF50',
    color: '#fff',
    padding: '10px',
    border: 'none',
    borderRadius: '5px',
    fontSize: '16px',
    cursor: 'pointer',
  },
  toggleButton: {
    marginTop: '10px',
    backgroundColor: 'transparent',
    color: '#007BFF',
    border: 'none',
    cursor: 'pointer',
    fontSize: '14px',
  },
  whatsappBox: {
    marginBottom: '20px',
    backgroundColor: '#e6f7f1',
    border: '1px solid #25d366',
    borderRadius: '8px',
    padding: '12px',
  },
  whatsappText: {
    marginBottom: '10px',
    color: '#333',
    fontSize: '14px',
  },
  whatsappButton: {
    backgroundColor: '#25d366',
    color: 'white',
    padding: '10px 15px',
    textDecoration: 'none',
    borderRadius: '6px',
    fontWeight: 'bold',
    display: 'inline-block',
  },
};

export default AuthForm;

File: flutter_app/lib/widgets/auth_form.dart
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert'; // For JSON decoding
import 'package:url_launcher/url_launcher.dart';

class AuthForm extends StatefulWidget {
  const AuthForm({super.key});
  @override
  State<AuthForm> createState() => _AuthFormState();
}

class _AuthFormState extends State<AuthForm> {
  bool isLogin = true;
  bool isLoggedIn = false;

  final _formKey = GlobalKey<FormState>();
  String name = '';
  String email = '';
  String phone = '';

  // Change this to your backend URL (use your machine IP if testing on physical device)
  final String baseUrl = 'http://localhost:5000';

  // Future<void> submit() async {
  //   if (!_formKey.currentState!.validate()) return;
  //   _formKey.currentState!.save();

  //   String finalPhone = phone.trim();
  //   if (!finalPhone.startsWith('+')) {
  //     finalPhone = '+91$finalPhone';
  //   }

  //   final url = Uri.parse('$baseUrl/api/auth/${isLogin ? 'login' : 'signup'}');

  //   Map<String, String> payload = isLogin
  //       ? {'phone': finalPhone}
  //       : {'name': name, 'email': email, 'phone': finalPhone};

  //   try {
  //     final response = await http.post(url, body: payload);
  //     if (response.statusCode == 200) {
  //       final data = jsonDecode(response.body);
  //       ScaffoldMessenger.of(context)
  //           .showSnackBar(SnackBar(content: Text(data['msg'] ?? 'Success')));
  //       if (isLogin) {
  //         setState(() {
  //           isLoggedIn = true;
  //         });
  //       }
  //     } else {
  //       ScaffoldMessenger.of(context).showSnackBar(
  //           SnackBar(content: Text('Error: ${response.statusCode}')));
  //     }
  //   } catch (e) {
  //     ScaffoldMessenger.of(context)
  //         .showSnackBar(SnackBar(content: Text('Error: $e')));
  //   }
  // }
  Future<void> submit() async {
  if (!_formKey.currentState!.validate()) return;
  _formKey.currentState!.save();

  String finalPhone = phone.trim();
  if (!finalPhone.startsWith('+')) {
    finalPhone = '+91$finalPhone';
  }

  final url = Uri.parse('$baseUrl/api/auth/${isLogin ? 'login' : 'signup'}');

  Map<String, String> payload = isLogin
      ? {'phone': finalPhone}
      : {'name': name, 'email': email, 'phone': finalPhone};

  try {
    final response = await http.post(
      url,
      headers: {'Content-Type': 'application/json'},  // <-- Important!
      body: jsonEncode(payload),                       // <-- Important!
    );

    if (response.statusCode == 200 || response.statusCode == 201) {
      final data = jsonDecode(response.body);
      ScaffoldMessenger.of(context)
          .showSnackBar(SnackBar(content: Text(data['msg'] ?? 'Success')));
      if (isLogin) {
        setState(() {
          isLoggedIn = true;
        });
      }
    } else {
      ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Error: ${response.statusCode}')));
    }
  } catch (e) {
    ScaffoldMessenger.of(context)
        .showSnackBar(SnackBar(content: Text('Error: $e')));
  }
}


  @override
  Widget build(BuildContext context) {
    if (isLoggedIn) {
      return Scaffold(
        appBar: AppBar(title: const Text('Welcome')),
        body: Center(
          child: Column(
            mainAxisAlignment: MainAxisAlignment.center,
            children: const [
              Text('🎉 Welcome to Surgy Innovation Labs',
                  style: TextStyle(fontSize: 22)),
              SizedBox(height: 10),
              Text("We're excited to have you on board."),
            ],
          ),
        ),
      );
    }

    return Scaffold(
      appBar: AppBar(
          title: Text(isLogin ? 'Login - Surgy Innovations Lab' : 'Signup - Surgy Innovations Lab')),
      body: Padding(
        padding: const EdgeInsets.all(16),
        child: Form(
          key: _formKey,
          child: ListView(
            children: [
              if (!isLogin) ...[
                TextFormField(
                  decoration: const InputDecoration(labelText: 'Name'),
                  validator: (val) =>
                      val == null || val.isEmpty ? 'Enter your name' : null,
                  onSaved: (val) => name = val ?? '',
                ),
                TextFormField(
                  decoration: const InputDecoration(labelText: 'Email'),
                  validator: (val) =>
                      val == null || !val.contains('@') ? 'Enter valid email' : null,
                  onSaved: (val) => email = val ?? '',
                ),
                const SizedBox(height: 10),
                Container(
                  padding: const EdgeInsets.all(12),
                  decoration: BoxDecoration(
                      color: Colors.green.shade100,
                      border: Border.all(color: Colors.green),
                      borderRadius: BorderRadius.circular(8)),
                  child: Column(
                    children: [
                      const Text(
                        '📢 Before signing up: To receive WhatsApp alerts, you must register your number.',
                        style: TextStyle(fontSize: 14),
                      ),
                      const SizedBox(height: 8),
                      TextButton(
                        style: TextButton.styleFrom(
                          backgroundColor: Colors.green,
                          foregroundColor: Colors.white,
                          padding: const EdgeInsets.symmetric(vertical: 10, horizontal: 16),
                          shape: RoundedRectangleBorder(
                              borderRadius: BorderRadius.circular(6)),
                        ),
                        onPressed: () async {
                          final url = Uri.parse(
                              'https://wa.me/14155238886?text=join%20eager-forest');
                          if (!await launchUrl(url)) {
                            ScaffoldMessenger.of(context).showSnackBar(
                              const SnackBar(content: Text('Could not launch WhatsApp')),
                            );
                          }
                        },
                        child: const Text('✅ Click here to register on WhatsApp'),
                      ),
                    ],
                  ),
                ),
                const SizedBox(height: 20),
              ],
              TextFormField(
                decoration: const InputDecoration(
                    labelText: 'Phone (e.g., +91XXXXXXXXXX)'),
                keyboardType: TextInputType.phone,
                validator: (val) =>
                    val == null || val.isEmpty ? 'Enter phone number' : null,
                onSaved: (val) => phone = val ?? '',
              ),
              const SizedBox(height: 20),
              ElevatedButton(
                onPressed: submit,
                child: Text(isLogin ? 'Login' : 'Signup'),
              ),
              TextButton(
                onPressed: () {
                  setState(() {
                    isLogin = !isLogin;
                  });
                },
                child: Text('Switch to ${isLogin ? 'Signup' : 'Login'}'),
              )
            ],
          ),
        ),
      ),
    );
  }
}
